R/mz_summary.R

```diff
@@ -0,0 +1,171 @@
+#' data("x_test", package = "MetNet")
+#' transformation <- rbind(
+#'     c("Monosaccharide (-H2O)", "C6H10O5", "162.0528234315"),
+#'     c("Disaccharide (-H2O)", "C12H20O11", "340.1005614851"),
+#'     c("Trisaccharide (-H2O)", "C18H30O15", "486.1584702945"))
+#' transformation <- data.frame(group = transformation[, 1],
+#'                                 formula = transformation[, 2],
+#'                                 mass = as.numeric(transformation[, 3]))
+#' am_struct <- structural(x_test, transformation, ppm = 5, directed = TRUE)
+#' # unfiltered mz difference counts
+#' mz_summary(am_struct)
+#' # filtered mz difference counts 
+#' mz_summary(am_struct, filter = 2)
+#'
+#'@export
+#'
+#'@importFrom dplyr `%>%`
+mz_summary <- function(am, filter = 0){
+  
+  if (!is(am, "AdjacencyMatrix")) {
+    stop("'am' is not an 'AdjacencyMatrix' object")
+  }
+  
+  if (!validObject(am)) 
+    stop("'am' must be a valid 'AdjacencyMatrix' object") 
+  
+  am_df <- as.data.frame(am)
+  
+  if (!"combine" %in% type(am) & !"structural" %in% type(am) ) 
+    stop("'am' is not a type 'structural' or 'combine'")
+  
+  if (!1 %in% am_df$binary) 
+    stop("'am' does not contain any mass-differences")
+  
+  if (!is.numeric(filter)) 
+    stop("'filter' needs to be numeric")
+  
+  if (filter < 0) 
+    stop("'filter' needs to be 0 or positive numeric")
+  
+  # if AdjacencyMatrix from `combine` is used 
+  if("combine" %in% type(am)){
+    am_df <- am_df[am_df$combine_binary == 1, ]
+    am_df <- na.omit(am_df, "combine_binary")
+    
+    sum_mass <- am_df %>% 
+      dplyr::group_by(combine_mass_difference) %>%  
+      dplyr::summarise(count = dplyr::n()) %>%
+      as.data.frame()
+    
+    sum_transform <- am_df %>% 
+      dplyr::group_by(combine_transformation) %>% 
+      dplyr::summarise(count = dplyr::n()) %>%
+      as.data.frame()  %>% 
+      tibble::add_column(sum_mass$combine_mass_difference)
+  }
+  
+  # if AdjacencyMatrix from `structural` is used 
+  else {
+    am_df <- am_df[am_df$binary == 1, ]
+    
+    sum_mass <- am_df %>% 
+      dplyr::group_by(mass_difference) %>%  
+      dplyr::summarise(count = dplyr::n()) %>%
+      as.data.frame()
+    
+    sum_transform <- am_df %>% 
+      dplyr::group_by(transformation) %>% 
+      dplyr::summarise(count = dplyr::n()) %>%
+      as.data.frame()  %>% 
+      tibble::add_column(sum_mass$mass_difference)
+  }
+  
+  colnames(sum_transform) <- c("transformation", "counts", "mass_difference")
+  sum_transform <- sum_transform %>% 
+    dplyr::select(transformation, mass_difference, counts)
+
+    sum_f <- sum_transform[sum_transform$counts >= filter, ]
+    return(sum_f)
+    
+  
+}
+
+
+
+
+#' @name mz_vis
+#'
+#' @title Visualize mass-difference distribution 
+#'
+#' @description
+#' The function `mz_vis` visualizes the mass-difference distribution,
+#' which has been summarized by `mz_summary`. 
+#' 
+#' @param x
+#' `list`, previously generated by `mz_summary`. Needs to contain
```

Comment:
one small note here that I just noticed. `mz_summary` returns a `data.frame`. I guess it would also be better to rename `list` to `data.frame`, use `df` or something similar instead of `x`  and check within the function that `df` is a `data.frame`:

    if (!is.data.frame(df)) stop(...)

Reply:
Woops! You are absolutely right. I think it was a list some time ago. I will change that.